### PR TITLE
memory/alloc: Use SCCalloc instead of malloc/memset

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #9305 

Replace `SCMalloc/memset` with `SCCalloc`

There are many places that use the following pattern:
```
     mem = SCMalloc(sizeof(foo));
     if (mem) {
        memset(&mem, 0, sizeof(foo));
    }
```

These can all be replaced by using `SCCalloc` which returns 0-filled memory.

Describe changes:
- Changed `SCMalloc` followed by `memset` to `SCCalloc` 

Updates:
- Rebase

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 15602

File: src/tm-threads.c [hunk 1]
```diff
@@ -913,10 +912,9 @@ ThreadVars *TmThreadCreate(const char *name, const char *inq_name, const char *i
     SCLogDebug("creating thread \"%s\"...", name);
 
     /* XXX create separate function for this: allocate a thread container */
```
Comment (jufajardini) [inline src/tm-threads.c:914]: nit/curious: is this comment still valid? (wondering if we could clean it up)

File: src/output-json-http2.c [hunk 2]
```diff
@@ -153,7 +153,7 @@ static OutputInitResult OutputHttp2LogInitSub(ConfNode *conf, OutputCtx *parent_
     OutputInitResult result = { NULL, false };
     OutputJsonCtx *ojc = parent_ctx->data;
 
-    OutputHttp2Ctx *http2_ctx = SCMalloc(sizeof(OutputHttp2Ctx));
+    OutputHttp2Ctx *http2_ctx = SCCalloc(1, sizeof(OutputHttp2Ctx));
```
Comment (jufajardini) [inline src/output-json-http2.c:156]: Curious: I've noticed that here and in some other places you've replaced SCMalloc even though there's no `memset` following. (`output-pgsql`, `output-smtp`), and I've also noticed that there are still SCMalloc's left. Is there a reason for replacing these specific ones, and not others?

**Review by jufajardini — COMMENTED** [review]
Going over this, had some questions in mind, left them in inline comments.

File: src/util-debug-filters.c [hunk 3]
```diff
@@ -925,20 +920,18 @@ void SCLogAddToFGFFuncList(SCLogFGFilterFile *fgf_file,
...
     SCLogFGFilterLine *fgf_line_temp = NULL;
 
-    if ( (fgf_func_temp = SCMalloc(sizeof(SCLogFGFilterFunc))) == NULL) {
+    if ((fgf_func_temp = SCCalloc(1, sizeof(SCLogFGFilterFunc))) == NULL) {
         FatalError("Fatal error encountered in SCLogAddToFGFFuncList. Exiting...");
     }
-    memset(fgf_func_temp, 0, sizeof(SCLogFGFilterFunc));
 
     if ( function != NULL && (fgf_func_temp->func = SCStrdup(function)) == NULL) {
         printf("Error Allocating memory\n");
```
[the remark this answers is not in the document]
Comment (jlucovsky) [reply src/util-debug-filters.c:928]: I'd suggest a separate ticket for this. I've got a few ideas queued for 8 and we can tackle the `printf` calls then?

File: src/output-json-http2.c (hunk 2 above)
Comment (jlucovsky) [reply src/output-json-http2.c:156]: Good question.

There are some places where a `malloc` is done and then the entire structure is populated (so no zeroing is needed). There are other places where a `calloc` should've been but was using `malloc`

That said, if I've missed anything or was too aggressive, please let me know

**Comment by jlucovsky** [comment]
Fixed up issues with rebase -- continued in #9388

File: src/output-json-http2.c (hunk 2 above)
Comment (jufajardini) [reply src/output-json-http2.c:156]: Thanks for taking the time to explain, I haven't seen any place where the changes seemed out of place, so all good in that sense :)

**Pull request closed without merging**